tla-types/src/test/scala/at/forsyte/apalache/tla/typecomp/BuilderTest.scala

```diff
@@ -31,6 +31,157 @@ import scala.collection.immutable.SortedMap
  *   - Tests whether a TBuilderInstruction, which is supposed to construct a well-typed operator expression actually
  *     does
  *   - Tests whether all of the inputs which would have produced an ill-typed expression actually cause `build` to fail
+ *
+ * <h1>HOW TO WRITE A NEW TEST</h1>
+ *
+ * Suppose we want to add a test for TLA+ operator (binary, in this example), `Foo(x,y) == ...` which has an internal
+ * representation of `foo` and is built by the `builder.foo` method. Assume, for the purposes of this example, that Foo
+ * takes two arguments - `x: a` and `y: a -> b` - and returns a value of type `b`, for any pair of types `a` and `b`.
+ *
+ * We add a test("foo") to TestOperCategoryBuilder, where OperCategory is the class of operators foo belongs to
+ * (TlaSetOper, ApalacheOper, etc.). Our goal is to use [[checkRun]] to test `builder.foo` on inputs of varying types.
+ * To do this, we need to determine the following:
+ *
+ * <ol>
+ *
+ * <li> The number of arguments to `builder.foo` and their types. In our example, `builder.foo` takes two
+ * [[TBuilderInstruction]] arguments. Typically, we introduce a type alias, so in our case
```

Comment:
In most cases, though some builder methods take string or integer literals, or even `TlaType1` hints.

Reply:
What's the purpose of introducing these type aliases? And where do these get introduced?

Reply:
I see! I suggest expanding the final example to include an entire test, along with the type aliases and the "run" invocation etc. I feel like that would really help bring everything together. Thanks for explaining.

Reply:
How is this file not exactly that?
https://github.com/informalsystems/apalache/blob/9346302d6c89fcc7ea13667a23fdc80c74795957/tla-types/src/test/scala/at/forsyte/apalache/tla/typecomp/HOWTOTestFooBuilder.scala

Reply:
https://github.com/informalsystems/apalache/blob/9346302d6c89fcc7ea13667a23fdc80c74795957/tla-types/src/test/scala/at/forsyte/apalache/tla/typecomp/BuilderTest.scala#L184

Reply:
lol, bad reading on my part! sorry for the confusion here!

Reply:
> https://github.com/informalsystems/apalache/blob/9346302d6c89fcc7ea13667a23fdc80c74795957/tla-types/src/test/scala/at/forsyte/apalache/tla/typecomp/BuilderTest.scala#L184

I think it would help to lift this sentence to the top of the HOWTO, so one can follow along.